### PR TITLE
bootstrap: update defaults for `compiler` and `library` aliases

### PR DESCRIPTION
**Pull request by lukas-code**
* `x doc compiler` now documents all of compiler, not just `rustc_driver`.
* `x doc` with compiler docs enabled now includes `rustc-main` and `rustc_smir`. `rustc_codegen_llvm` is only included if the LLVM backend is enabled, which is the default.
* `x doc library` now excludes `sysroot`.
* `x check compiler` and `x check library` now properly check tests/benches/examples of all compiler or library crates, respectively. Note that `x check compiler` will check the library artifacts, but not tests.

fixes the fallout from https://github.com/rust-lang/rust/pull/111955, cc @jyn514

### CODE REVIEW

**Comment by jyn514** [comment]
cc https://github.com/rust-lang/rust/issues/112440

File: compiler/rustc_smir/src/lib.rs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 //! The WIP stable interface to rustc internals.
 //!
-//! For more information see https://github.com/rust-lang/project-stable-mir
+//! For more information see <https://github.com/rust-lang/project-stable-mir>
```
Comment (albertlarsan68) [inline compiler/rustc_smir/src/lib.rs:3]: Can you remove this change?

**Review by albertlarsan68 — COMMENTED** [review]
LGTM, r=me with this change

File: compiler/rustc_smir/src/lib.rs (hunk 1 above)
Comment (albertlarsan68) [reply compiler/rustc_smir/src/lib.rs:3]: Thanks for the explanation

**Comment by albertlarsan68** [comment]
Thanks for the PR!
@bors r+

**Comment by matthiaskrgr** [comment]
@bors r-
probably fails here https://github.com/rust-lang/rust/pull/113669#issuecomment-1634760956
feel free to reapprove if you think there this is not actually the cause

**Comment by lukas-code** [comment]
Yeah that rollup failure is definitely caused by this PR. It's more broken docs for `rustc_smir` that got revealed, because this PR actually builds the docs now.

**Comment by albertlarsan68** [comment]
@oli-obk, @celinval What do you think about this failure? Do not merge this PR or fix the docs?

**Comment by jyn514** [comment]
certainly fix the docs

**Pull request merged**